Advanced/One-Click Macros/Measure in Calc Item.csx

```diff
@@ -0,0 +1,216 @@
+};
+
+if (DynamicMeasureCG == null) { return; } // doesn't work in TE3 as cancel button doesn't return null in TE3
+
+
+
+//INIT DUMMY MEASURE (if necessary) 
+var dummyMeasureCandidates = DynamicMeasureCG.Measures.Where(x => x.GetAnnotation("@AgulloBernat") == dummyMeasureTag);
+
+Measure dummyMeasure = null as Measure; 
+
+
+if (dummyMeasureCandidates.Count() == 1) { 
+    
+    dummyMeasure = dummyMeasureCandidates.First() as Measure; 
+
+} else if (dummyMeasureCandidates.Count() < 1) {
+    
+    string dummyMeasureName = Interaction.InputBox("Enter a name for the dummy measure", "Measure Name", "", 740, 400);
+
+    if(dummyMeasureName == "") {
+        Error("No name provided");         
+        return;
+    };
+    
+    //add dummy measure if not present yet
+    if(!DynamicMeasureCG.Measures.Where(m => m.Name == dummyMeasureName).Any())  {
+        
+        dummyMeasure = DynamicMeasureCG.AddMeasure(dummyMeasureName,"0");
+        
+    } else { 
+
+        //TODO: ASK FOR CONFIRMATION 
+
+        //get the reference if already exists 
+        dummyMeasure = DynamicMeasureCG.Measures.Where(m => m.Name == dummyMeasureName).First(); 
+    };
+
+    //in any case add the annotation so it will be found next time 
+    dummyMeasure.SetAnnotation("@AgulloBernat",dummyMeasureTag); 
+
+
+} else{ 
+    //very weird! already two dummy measures? 
+    
+    dummyMeasure = 
+        SelectMeasure(
+            dummyMeasureCandidates,
+            label:"Select your Dummy Measure for your Dynamic Measure Calculation Group For Arbitrary 2-row Header"
+        ) as Measure; 
+
+
+} ; 
+
+foreach (var m in Selected.Measures) { 
+    
+    if (!DynamicMeasureCG.CalculationItems.Where(x => x.Name == m.Name).Any()) { 
+        string newCalcItemName = m.Name; 
+        string newCalcItemExpression = 
+             "IF(" + 
+             "    ISSELECTEDMEASURE( [" + dummyMeasure.Name + "] )," + 
+             "    VAR currentCalcItem =" + 
+             "        SELECTEDVALUE( "+  selectedCalcItemsCalcGroupColumn.DaxObjectFullName +", \"NO SELECTION\" )" + 
+             "    RETURN" + 
+             "        IF( currentCalcItem IN " + selectedCalcItems + ", [" + m.Name + "] )," + 
+             "    SELECTEDMEASURE()" + 
+             ")";
+        
+        CalculationItem newCalcItem =  DynamicMeasureCG.AddCalculationItem(newCalcItemName, newCalcItemExpression); 
+        newCalcItem.FormatDax(); 
+    } else { 
+        Info("Calculation item " + m.Name + " already present. Please modify manually, or delete it and try again"); 
+    };
+
+}; 
+
+
+//se asegura que el calculation group original siga teniendo precedencia superior 
+//int DynamicMeasureCGprecedence = DynamicMeasureCG.CalculationGroup.Precedence;
+//int SelectedCalculationGroupPrecedence = selectedCalculationGroup.CalculationGroup.Precedence; 
+
+
+
+
+
+
+
```

Comment:
Commented-out code in production scripts should be:

A. For example/instruction/documentation purposes
B. Optional functionality, disabled by default, which the user can choose to enable by un-commenting it
C. Removed

If this commented-out code block fits the A and/or B criteria above, please add a new comment on the line above it explaining its purpose, and how to use it (if applicable). Otherwise, please remove it. Thanks!